# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmModuleTypes.h
# Commit: 339e37a04ca1
# Full Hash: 339e37a04ca17a209114f9d346751e6d05f6a37d
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Preliminaries. r=yury
#   
#   Scaffolding: Add assertions everywhere execution currently depends on
#   a memory being a memory32.  These assertions will disappear by and by.
#   
# ==============================================================================

diff -r 83e52246d0ea -r 339e37a04ca1 js/src/wasm/WasmModuleTypes.h
--- a/js/src/wasm/WasmModuleTypes.h	Mon Oct 18 09:58:13 2021 +0000
+++ b/js/src/wasm/WasmModuleTypes.h	Mon Oct 18 09:58:13 2021 +0000
@@ -517,6 +517,11 @@
     return limits.initial * PageSize;
   }
 
+  uint64_t initialLength64() const {
+    MOZ_ASSERT(indexType() == IndexType::I64);
+    return limits.initial * PageSize;
+  }
+
   MemoryDesc() = default;
   explicit MemoryDesc(Limits limits) : limits(limits) {}
 };